rawler/src/decoders/dng.rs

```diff
@@ -401,19 +401,23 @@ impl<'a> DngDecoder<'a> {
 
     let mut read_matrix = |cal: DngTag, mat: DngTag| -> Result<()> {
       if let Some(c) = self.tiff.get_entry(mat) {
-        let illuminant: Illuminant = fetch_tiff_tag!(self.tiff, cal).force_u16(0).try_into()?;
+        let illuminant_val = self.tiff.get_entry(cal).map(|e| e.force_u16(0)).unwrap_or(21);
+        let illuminant: Illuminant = illuminant_val.try_into().unwrap_or(Illuminant::D65);
+
         let mut matrix = FlatColorMatrix::new();
         for i in 0..c.count() as usize {
           matrix.push(c.force_f32(i));
         }
-        assert!(matrix.len() <= 12 && !matrix.is_empty());
-        result.insert(illuminant, matrix);
+
+        if !matrix.is_empty() && matrix.len() <= 12 {
+            result.insert(illuminant, matrix);
+        }
       }
       Ok(())
     };
 
-    read_matrix(DngTag::CalibrationIlluminant1, DngTag::ColorMatrix1)?;
-    read_matrix(DngTag::CalibrationIlluminant2, DngTag::ColorMatrix2)?;
+    let _ = read_matrix(DngTag::CalibrationIlluminant1, DngTag::ColorMatrix1);
```

Comment:
This drops any error in read_matrix(), even errors maybe added/returned in future.
You've already eliminated all errors in read_matrix(), just leave this code untouched.